### ISSUE TITLE
reduce CPU usage

### ISSUE THREAD

**Issue by correabuscar** [MrTanoshii/rusty-autoclicker#45]
On Windows even if auto-clicking isn't enabled, due to egui/eframe it seems, there's something like 6% CPU usage(on 12 cores cpu (6+6)) constantly(more or less) but which noticeably turns on CPU fans to audible sounds for me, compared to when rust-autoclicker isn't running at all.

this probably cannot be fixed... i don't know.

**Comment by MrTanoshii** [comment]
I see what you mean

Tested quickly using the trusty windows task manager on a i7-7500U CPU

Unfocus:
- Low: 1%
- Avg: 4%
- High: 8%

Focused (Idle & autoclicking):
- Low: 28%
- Avg: 31%
- High: 32%

Seems to be an issue of the app doing too many things when focused, I'll look into slowing things down when not autoclicking. Maybe the refresh rate is too high?

**Comment by correabuscar** [comment]
I have noticed that this https://github.com/emilk/eframe_template  doesn't use any CPU at all(on Linux), so maybe it is possible to reduce the frame rate as you say. Or maybe somehow only update when needed, although, you're showing the current mouse coords in real time.

My problem was mostly when unfocused and not autoclicking.

**Comment by MrTanoshii** [comment]
> I can add Tracy profiling after eframe upgrades to a Tracy‑compatible wgpu (the current version is not). Let me know if you want me to handle it.
> 
> Note: egui’s immediate mode UI will always consume more CPU than a retained mode UI.

@Kumpelinus I'm not familiar with Tracy. Could it be integrated as an optional feature? Ideally, it would be available in local release builds but disabled in GitHub release builds. I'd love to have proper benchmarks, but not at the expense of regular users.

**Comment by Kumpelinus** [comment]
> > I can add Tracy profiling after eframe upgrades to a Tracy‑compatible wgpu (the current version is not). Let me know if you want me to handle it.
> > Note: egui’s immediate mode UI will always consume more CPU than a retained mode UI.
> 
> [@Kumpelinus](https://github.com/Kumpelinus) I'm not familiar with Tracy. Could it be integrated as an optional feature? Ideally, it would be available in local release builds but disabled in GitHub release builds. I'd love to have proper benchmarks, but not at the expense of regular users.

@MrTanoshii Yes, tracy is set up to only be included in development environment in the majority of cases. It wouldn't affect any release builds that way.
